(torchelastic) make --max_restarts explicit in the quickstart and runner docs (#65675)

kiukchung opened the pull request:
Summary:
closes https://github.com/pytorch/pytorch/pull/65675

The default `--max_restarts` for `torch.distributed.run` was changed to `0` from `3` to make things backwards compatible with `torch.distributed.launch`. Since the default `--max_restarts` used to be greater than `0` we never documented passing `--max_restarts` explicitly in any of our example code.

Test Plan: N/A doc change only

Differential Revision: D31279544



cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

H-Huang commented on docs/source/elastic/quickstart.rst:
```diff
@@ -22,6 +23,7 @@ and at most ``MAX_SIZE`` nodes.
     torchrun
         --nnodes=MIN_SIZE:MAX_SIZE
         --nproc_per_node=TRAINERS_PER_NODE
+        --max_restarts=NUM_ALLOWED_MEMBERSHIP_CHANGE
```
should it be `NUM_ALLOWED_MEMBERSHIP_CHANGE_OR_FAILURES`? or is that too verbose

kiukchung replied:
yea, you're right updated it to `NUM_ALLOWED_FAILRES_OR_MEMBERSHIP_CHANGES` in V2

H-Huang reviewed (approved):
LGTM!

Commit:
(torchelastic) make --max_restarts explicit in the quickstart and runner docs (#65838)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/65838

closes https://github.com/pytorch/pytorch/pull/65675

The default `--max_restarts` for `torch.distributed.run` was changed to `0` from `3` to make things backwards compatible with `torch.distributed.launch`. Since the default `--max_restarts` used to be greater than `0` we never documented passing `--max_restarts` explicitly in any of our example code.

Test Plan: N/A doc change only

Reviewed By: d4l3k

Differential Revision: D31279544

fbshipit-source-id: afabb4ac492ba0640752efbc8e36ed485d97ec5a
